File: MyAnimeList/src/main/java/com/example/myanimelist/di/modules/UsersRepositoryModule.java
package com.example.myanimelist.di.modules;

import com.example.myanimelist.repositories.users.IUsersRepository;
import com.example.myanimelist.repositories.users.UsersRepository;
import dagger.Binds;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;
@Module
public interface UsersRepositoryModule {
    @Singleton
    @Binds
    IUsersRepository bindUserRepository(UsersRepository repository);
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/di/modules/UsersRepositoryModule.kt
package com.example.myanimelist.di

import com.example.myanimelist.repositories.users.IUsersRepository
import com.example.myanimelist.repositories.users.UsersRepository
import dagger.Binds
import dagger.Module
import javax.inject.Singleton

@Module
interface UsersRepositoryModule {
    @Singleton
    @Binds
    fun bindUsersRepository(repository: UsersRepository) : IUsersRepository
}